### DISCUSSION TITLE
Improve user interactions

### DISCUSSION THREAD

**Discussion by s3rrot** [boken-engine/boken-engine#94] [Ideas]
We could add new gestures for iOS version and mouse+keyboard interaction for [desktop version](https://github.com/boken-engine/boken-engine/discussions/91).

**Reference:**
-  [iOS Human Interface Guidelines - Gestures](https://developer.apple.com/design/human-interface-guidelines/ios/user-interaction/gestures/)
- [macOS Human Interface Guidelines - Keyboard](https://developer.apple.com/design/human-interface-guidelines/macos/user-interaction/keyboard/)
- [macOS Human Interface Guidelines - Mouse and Trackpad](https://developer.apple.com/design/human-interface-guidelines/macos/user-interaction/mouse-and-trackpad/)

**Comment by yeraydavidrodriguez** [comment]
Actually, Boken already has support for swipe gestures that can be set using the GestureManager class. This works this way: Right after the scene is loaded, the host app can call createNavigationGestures to set up the gestures. When a scene has the navigation options set to next, back or both, the user can change to adjacent scenes using swipes.
Keyboard or mouse support is not explicitly supported by Boken Engine, but as the branch buttons are implemented using a sprinteNode that has handlers for both click and touch, at least the mouse-based interaction seems to be provided out-of-the-box.
